cider/tests/utils/DuckDbQueryRunner.cpp

```diff
@@ -532,7 +532,25 @@ DuckDbResultConvertor::fetchDataToArrowFormattedCiderBatch(
     auto chunk = result->Fetch();
     if (!chunk || (chunk->size() == 0)) {
       if (batch_res.empty()) {
-        batch_res.push_back(std::make_shared<CiderBatch>());
+        auto arrow_array = CiderBatchUtils::allocateArrowArray();
+        auto arrow_schema = CiderBatchUtils::allocateArrowSchema();
+
+        arrow_array->length = 0;
+        arrow_array->n_children = 0;
+        arrow_array->buffers = nullptr;
+        arrow_array->n_buffers = 0;
+        arrow_array->private_data = nullptr;
+        arrow_array->children = (ArrowArray**)std::malloc(0);
```

Comment:
what does malloc(0) returns ? and we should avoid call malloc directly.

Reply:
I think we will get a valid pointer by malloc(0) instead of nullptr. 

Went through all the places using `getChildrenArrayPtr()` and it looks okay if we just set it to nullptr.